From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 9 May 2020 14:42:37 +0200
Subject: Change default webRTC policy to not use any address

This prevents leakage of the local IP address.

See also: https://github.com/bromite/bromite/issues/553
---
 chrome/browser/ui/browser_ui_prefs.cc                      | 4 ++--
 .../peerconnection/peer_connection_dependency_factory.cc   | 7 ++++---
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/chrome/browser/ui/browser_ui_prefs.cc b/chrome/browser/ui/browser_ui_prefs.cc
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -87,8 +87,8 @@ void RegisterBrowserUserPrefs(user_prefs::PrefRegistrySyncable* registry) {
                                 false);
 #endif
   // TODO(guoweis): Remove next 2 options at M50.
-  registry->RegisterBooleanPref(prefs::kWebRTCMultipleRoutesEnabled, true);
-  registry->RegisterBooleanPref(prefs::kWebRTCNonProxiedUdpEnabled, true);
+  registry->RegisterBooleanPref(prefs::kWebRTCMultipleRoutesEnabled, false);
+  registry->RegisterBooleanPref(prefs::kWebRTCNonProxiedUdpEnabled, false);
   registry->RegisterStringPref(prefs::kWebRTCIPHandlingPolicy,
                                blink::kWebRTCIPHandlingDefault);
   registry->RegisterStringPref(prefs::kWebRTCUDPPortRange, std::string());
diff --git a/third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.cc b/third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.cc
--- a/third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.cc
+++ b/third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.cc
@@ -421,12 +421,13 @@ PeerConnectionDependencyFactory::CreatePortAllocator(
           port_config.enable_nonproxied_udp = false;
           break;
         case DEFAULT:
-          port_config.enable_multiple_routes = true;
-          port_config.enable_nonproxied_udp = true;
+          port_config.enable_multiple_routes = false;
+          port_config.enable_nonproxied_udp = false;
+          port_config.enable_default_local_candidate = false;
           break;
       }
 
-      VLOG(3) << "WebRTC routing preferences: "
+      LOG(INFO) << "WebRTC routing preferences: "
               << "policy: " << policy
               << ", multiple_routes: " << port_config.enable_multiple_routes
               << ", nonproxied_udp: " << port_config.enable_nonproxied_udp
-- 
2.17.1

